llap-tez/src/java/org/apache/hadoop/hive/llap/tezplugins/LlapTaskCommunicator.java

```diff
@@ -1163,4 +1165,15 @@ private QueryIdentifierProto constructQueryIdentifierProto(int dagIdentifier) {
   public String getAmHostString() {
     return amHost;
   }
+
+  /**
+   * Overrides TezTaskCommunicatorImpl.getTotalUsedMemory in order to provide correct aggregated memory usage.
+   * In LLAP, every container reports the whole used heap of the daemon they're running in, so we need to consider
+   * every usedMemory once per daemon.
+   * @return
+   */
+  @Override
+  public long getTotalUsedMemory() {
+    return pingedNodeMap.values().stream().mapToLong(c -> c.usedMemory).sum();
```

Comment:
Sum does not provide much value here. Maybe we could print it as an average percentage across daemons, but having some kind of percentile here could actually reveal if some daemons face more pressure than others

Reply:
i.e., could print something like: Avg usage percent: 50%. 80th percentile: 80%, 90th percentile: 99%

Reply:
thanks for the comments @pgaref 
```
Sum does not provide much value here
```
I see, the sum is just one information (we need at least the number of daemons additionally to make it have a meaning), but the percentage looks more useful
I was already thinking about percentage, what should be the value of 100%? I mean, usedMemory is heap usage in daemons, what do you think we should consider as the "whole" memory?
I found adjustedExecutorMemory which would make sense:
https://github.com/apache/hive/blob/1046f41ea36ab3c8b036481128ba9b76dda2882a/llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/LlapDaemon.java#L227

Reply:
also, I need to plan this consciously, because every change needs to flow through tez DagClient API, which is protobuf based and doesn't support map<> with the current protobuf version, so I need to find out another way to propagate arbitrary memory-related data